libs/tokens/src/services/fetchTokenList.ts

```diff
@@ -1,16 +1,20 @@
-import { MAINNET_PROVIDER } from '@cowprotocol/common-const'
+import { RPC_URLS } from '@cowprotocol/common-const'
 import {
   contenthashToUri,
   isAddress,
   parseENSAddress,
   resolveENSContentHash,
   uriToHttp,
 } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { JsonRpcProvider } from '@ethersproject/providers'
 import { TokenList } from '@uniswap/token-lists'
 
 import { ListSourceConfig, ListState } from '../types'
 import { validateTokenList } from '../utils/validateTokenList'
 
+const MAINNET_PROVIDER = new JsonRpcProvider(RPC_URLS[SupportedChainId.MAINNET])
```

Comment:
Side topic: so even when we are connected to a different chain, we always have a mainnet provider?

This can make the ENS query thingy more easy to implement: https://github.com/cowprotocol/cowswap/blob/develop/libs/ens/src/hooks/useENSRegistrarContract.ts#L15

Reply:
Yes, we consume ENS only from mainnet.
One day we need to refactor ENS related stuff.

Reply:
I meant, ENS should work on all chains, but the main registry is only on mainnet.